test/util.py

```diff
@@ -117,6 +119,105 @@ def hess(_):
                          max_sensi_order, integrated, x)
 
 
+class CRProblem:
+    """ODE model of a conversion reaction A <-> B.
+
+    Uses automatic differentiation for derivative calculation.
+    """
+
+    def __init__(
+        self, n_t: int = 10, max_t: float = 15., x0: anp.ndarray = None,
+        p_true: anp.ndarray = None, sigma: float = 0.02,
+        lb: anp.ndarray = None, ub: anp.ndarray = None,
+    ):
+        self.ts = anp.linspace(0, max_t, n_t)
+        if x0 is None:
+            x0 = anp.array([1., 0.])
+        self.x0: anp.ndarray = x0
+        if p_true is None:
+            p_true = anp.array([0.06, 0.08])
+        self.p_true: anp.ndarray = p_true
+        self.sigma: float = sigma
+        if lb is None:
+            lb = anp.array([0., 0.])
+        self.lb = lb
+        if ub is None:
+            ub = anp.array([0.5, 0.5])
+        self.ub = ub
+
+        y_true = self.get_fy()(self.p_true)
+        # seed random number generator for reproducibility
+        rng = anp.random.Generator(anp.random.PCG64(0))
+        self.data = y_true + sigma * rng.normal(size=y_true.shape)
+
+    def get_fy(self):
+        """Systems states, fully observed, analytic solution."""
```

Comment:
Could you maybe state the ODE somewhere in the doc strings?

Reply:
done